node/libs/storage/src/types.rs

```diff
@@ -56,6 +53,20 @@ impl DatabaseKey {
     }
 }
 
+/// A payload of a proposed block which is not known to be finalized yet.
+/// Replicas have to persist such proposed payloads for liveness:
+/// consensus may finalize a block without knowing a payload in case of reproposals.
+/// TODO(gprusak): we do not store the BlockHeader, because it is always
+///   available in the LeaderPrepare message. However for better
+///   visibility we might want to store it anyway.
```

Comment:
```suggestion
/// Currently we do not store the BlockHeader, because it is always
///   available in the LeaderPrepare message.
```
Let's avoid adding any new TODOS. If it's not important enough to discuss with the team and/or create a Linear ticket, then it's not important.